cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageDatasetTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.metadata.lineage.field;
+
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.lineage.field.EndPoint;
+import co.cask.cdap.api.lineage.field.InputField;
+import co.cask.cdap.api.lineage.field.Operation;
+import co.cask.cdap.api.lineage.field.ReadOperation;
+import co.cask.cdap.api.lineage.field.TransformOperation;
+import co.cask.cdap.api.lineage.field.WriteOperation;
+import co.cask.cdap.common.app.RunIds;
+import co.cask.cdap.data2.datafabric.dataset.DatasetsUtil;
+import co.cask.cdap.data2.dataset2.DatasetFrameworkTestUtil;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.id.DatasetId;
+import co.cask.cdap.proto.id.ProgramId;
+import co.cask.cdap.proto.id.ProgramRunId;
+import org.apache.tephra.TransactionAware;
+import org.apache.tephra.TransactionExecutor;
+import org.apache.twill.api.RunId;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Test for storage and retrieval of the field lineage operations.
+ */
+public class FieldLineageDatasetTest {
+  @ClassRule
+  public static DatasetFrameworkTestUtil dsFrameworkUtil = new DatasetFrameworkTestUtil();
+
+  @Test
```

Comment:
I think this could use more test cases. You need to test 
- that the time ranges work as designed (start inclusive, end exclusive)
- edge cases such as cyclic graphs, graphs where the writes or the reads are not connected, etc
- merging of the field summaries from multiple runs
- etc.

Reply:
Added tests for time range and merge of field summaries. Have already filed JIRA - https://issues.cask.co/browse/CDAP-13548 for handling cases of cycles, will add tests for it as a part of that JIRA.